packages/ui/src/components/Link/Link.tsx

```diff
@@ -0,0 +1,75 @@
+import NextLink, { type LinkProps } from 'next/link';
+import type { HTMLProps, Ref, RefObject } from 'react';
+import { twMerge } from 'tailwind-merge';
+
+import { Icon } from './Icon';
+import { type BaseProps, LinkSize, LinkVariant } from './types';
+
+export function Link({
+    variation = 'primary',
+    className,
+    forwardRef,
+    forceRefresh,
+    size = 'default',
+    icon,
+    iconPlacement = 'left',
+    disabled,
+    children,
+    contentClassName,
+    localeCode,
+    href,
+    ...linkProps
+}: Link.Props) {
+    const Element = forceRefresh ? 'a' : NextLink;
+
+    const hrefWithLocale = localeCode ? `/${localeCode}/${href.toString()}` : href.toString();
+
+    return (
+        <Element
+            href={forceRefresh ? hrefWithLocale : href}
+            ref={forwardRef as any}
+            className={twMerge(
+                'flex items-center justify-center w-max rounded',
+                size === 'small' ? `label-medium` : `label-large`,
+                variation === 'primary' &&
+                    `
+  text-accent hover:text-accent-dark active:text-accent-dark focus:text-accent-dark focus-within:text-accent-dark
+  focus:ring-2 focus:ring-accent-lighter focus-within:ring-2 focus-within:ring-accent-lighter
+  ${disabled && 'text-accent-lighter pointer-events-none'}
+  `,
+                variation === 'secondary' &&
+                    `
+      text-gray-600 hover:text-gray-800 focus:text-gray-800 focus:ring-2 focus:ring-accent-lighter
+  focus-within:text-gray-800 focus-within:ring-2 focus-within:ring-accent-lighter active:text-gray-950 ${
+      disabled && 'text-gray-500 pointer-events-none'
+  }
+  `,
+                Boolean(Icon) && Boolean(children) && `gap-2`,
+                className,
+            )}
+            {...linkProps}
+        >
+            {iconPlacement === 'left' && <Icon icon={icon} />}
+            {/* If there are no children, we insert a zero-width space to preserve the line-height */}
+            <span className={contentClassName}>{children ?? <>&#8203;</>}</span>
```

Comment:
Based on our previous conversation, `children` should be a required prop, and this hack becomes redundant.